scripts/tests/test_generate_bad_phase37_artifacts.py

```
@@ -0,0 +1,156 @@
+    def test_generator_does_not_write_python_cache_files(self) -> None:
+        with tempfile.TemporaryDirectory() as tempdir:
+            temp = pathlib.Path(tempdir)
+            run_generator(temp / "mutations")
+            self.assertEqual(pycache_dirs(temp), [])
+
+    def test_generator_rejects_populated_output_directory(self) -> None:
+        with tempfile.TemporaryDirectory() as tempdir:
+            output_dir = pathlib.Path(tempdir) / "mutations"
+            output_dir.mkdir()
+            (output_dir / "stale.json").write_text("{}", encoding="utf-8")
+            completed = subprocess.run(
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)